src/dynarec/rv64/dynarec_rv64_emit_math.c

```diff
@@ -45,7 +45,9 @@ void emit_add32(dynarec_rv64_t* dyn, int ninst, rex_t rex, int s1, int s2, int s
             BEQZ(s5, 8);
             ORI(xFlags, xFlags, 1 << F_CF);
         } else {
-            ADD(s5, s1, s2);
+            AND(s3, s1, xMASK);
+            AND(s4, s2, xMASK);
+            ADD(s5, s3, s4);
```

Comment:
only 1 and with xMASK is needed actually...

Reply:
hmm why?

Reply:
s3 will have upper bit to 0. anything and that will still have the upper bit to 0.

Reply:
oh I think you misread line 50, it’s add, not and.

Reply:
ah right, add, not and :(